pkg/controller/clone-controller.go

```diff
@@ -190,10 +190,16 @@ func (r *CloneReconciler) Reconcile(req reconcile.Request) (reconcile.Result, er
 	_, nameExists := pvc.Annotations[AnnCloneSourcePod]
 	if !nameExists && sourcePod == nil {
 		pvc.Annotations[AnnCloneSourcePod] = createCloneSourcePodName(pvc)
+
+		// add finalizer before creating clone source pod
+		pvc = r.addFinalizer(pvc, cloneSourcePodFinalizer)
+
 		if err := r.updatePVC(pvc); err != nil {
 			return reconcile.Result{}, err
 		}
-		return reconcile.Result{Requeue: true}, nil
+
+		// will reconcile again after PVC update notification
+		return reconcile.Result{}, nil
```

Comment:
This may keep us from getting some of the "conflict errors seen in the controller log.